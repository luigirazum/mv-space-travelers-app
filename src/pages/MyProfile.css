.myProfile {
  margin-block: 1em;
  margin-inline: auto;
  display: flex;
  flex-wrap: wrap;
  gap: 5em;

  --profile-basis: 15rem;
}

.mySection {
  flex-grow: 1;
  flex-shrink: 0;
  flex-basis: var(--profile-basis);
}

.sectionTitle {
  font-size: var(--fs-super);
  color: var(--color-btn-primary);
}

.noItems {
  font-size: var(--fs-item-list);
  font-weight: var(--fw-bold);
  color: var(--color-btn-cancel);
  padding-inline-start: 1em;
  padding-block-start: 0.5em;
}

.noItems span {
  margin-inline: 0.5em;
}

.myList {
  margin-block: 0.5em;
  border: var(--border-profile);
  border-radius: 5px;
}

.myListItem {
  border-bottom: var(--border-profile);
  padding: 0.5em 1.25em;
  display: flex;
  align-items: center;
  --listItem-base: 7.75rem;
  transition: transform 0.2s ease-out, background-color 0.25s ease, border 0.2s ease-out;
}

.myListItem .btn {
  font-size: var(--fs-badge);
  margin-inline-start: auto;
  padding: 0.35em 0.75em;
  flex-grow: 1;
  flex-shrink: 0;
  flex-basis: var(--listItem-base);
}

.myListItem:last-child {
  border-bottom: none;
}

.myListItem:hover {
  background-color: var(--color-hoverRocket-hsl);
  transform: scale(1.05);
  border: var(--border-profile-hover);
  border-radius: 5px;
}

.myItemName {
  font-size: var(--fs-item-list);
  flex-grow: 13;
  flex-shrink: 0;
  flex-basis: var(--listItem-base);
}

.myListItem:hover .myItemName {
  font-weight: var(--fw-bold);
  color: var(--color-link-active);
}
