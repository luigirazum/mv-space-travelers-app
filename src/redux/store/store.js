import { configureStore } from '@reduxjs/toolkit';
import stateReducer from '../status/statusSlice';

const store = configureStore({
  reducer: {
    status: stateReducer,
  },
});

export default store;
