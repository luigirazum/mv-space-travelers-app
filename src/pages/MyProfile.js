import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { leaveMission } from '../redux/missions/missions';
import { cancelReservedRocket } from '../redux/rockets/rockets';
import wikiLogo from '../assets/imgs/logo/wikipedia_logo.png';
import './MyProfile.css';

const selectReservedRockets = (state) => {
  const { rockets: { inventory: rockets } } = state;
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);

  return reservedRockets;
};

const selectJoinedMissions = (state) => {
  const { missions: { avaliable: missions } } = state;
  const joinedMissions = missions.filter((mission) => mission.joined);

  return joinedMissions;
};

const MyProfile = () => {
  const myRockets = useSelector(selectReservedRockets);
  const myMissions = useSelector(selectJoinedMissions);
  const dispatch = useDispatch();

  const handleCancelReservedRocket = (id) => {
    dispatch(cancelReservedRocket(id));
  };

  const handleLeaveMission = (id) => {
    dispatch(leaveMission(id));
  };

  const myReservedRockets = myRockets.map((rocket) => {
    const { id, name, wikipedia } = rocket;

    return (
      <li key={id} className="myListItem">
        <p className="myItemName">
          {name}
        </p>
        <a
          href={wikipedia}
          target="_blank"
          rel="noreferrer noopener"
          className="moreInfo"
        >
          read more
          <span className="wikiLogo">
            <img src={wikiLogo} alt="wikipedia logo" />
          </span>
        </a>
        <button
          type="button"
          className="btn btn-cReserve"
          onClick={() => handleCancelReservedRocket(id)}
        >
          Cancel Reservation
        </button>
      </li>
    );
  });

  const myJoinedMissions = myMissions.map((mission) => {
    const { id, name, wikipedia } = mission;

    return (
      <li key={id} className="myListItem">
        <p className="myItemName">
          {name}
        </p>
        <a
          href={wikipedia}
          target="_blank"
          rel="noreferrer noopener"
          className="moreInfo"
        >
          read more
          <span className="wikiLogo">
            <img src={wikiLogo} alt="wikipedia logo" />
          </span>
        </a>
        <button
          type="button"
          className="btn btn-mLeave"
          onClick={() => handleLeaveMission(id)}
        >
          Leave Mission
        </button>
      </li>
    );
  });

  return (
    <>
      <div className="myProfile">
        <section className="mySection">
          <h2 className="sectionTitle">My Rockets</h2>
          { myReservedRockets.length
            ? (
              <ul className="myList">
                {myReservedRockets}
              </ul>
            )
            : (
              <p className="noItems">
                There are no Rockets that have been
                <span className="badge rktBadge resBadge">Reserved</span>
                by you.
              </p>
            )}
        </section>
        <section className="mySection">
          <h2 className="sectionTitle">My Missions</h2>
          { myJoinedMissions.length
            ? (
              <ul className="myList">
                {myJoinedMissions}
              </ul>
            )
            : (
              <p className="noItems">
                You haven&apos;t yet become an
                <span className="badge misBadge misActive">Active Member</span>
                of any Missions.
              </p>
            )}
        </section>
      </div>
    </>
  );
};

export default MyProfile;
