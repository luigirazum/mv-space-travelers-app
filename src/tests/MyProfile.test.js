import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/store/store';
import MyProfile from '../pages/MyProfile';

describe('My Profile Page component', () => {
  it('renders properly', () => {
    const tree = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
