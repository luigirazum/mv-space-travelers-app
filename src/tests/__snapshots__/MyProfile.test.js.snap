// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Profile Page component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="myProfile"
      >
        <section
          class="mySection"
        >
          <h2
            class="sectionTitle"
          >
            My Rockets
          </h2>
          <p
            class="noItems"
          >
            There are no Rockets that have been
            <span
              class="badge rktBadge resBadge"
            >
              Reserved
            </span>
            by you.
          </p>
        </section>
        <section
          class="mySection"
        >
          <h2
            class="sectionTitle"
          >
            My Missions
          </h2>
          <p
            class="noItems"
          >
            You haven't yet become an
            <span
              class="badge misBadge misActive"
            >
              Active Member
            </span>
            of any Missions.
          </p>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="myProfile"
    >
      <section
        class="mySection"
      >
        <h2
          class="sectionTitle"
        >
          My Rockets
        </h2>
        <p
          class="noItems"
        >
          There are no Rockets that have been
          <span
            class="badge rktBadge resBadge"
          >
            Reserved
          </span>
          by you.
        </p>
      </section>
      <section
        class="mySection"
      >
        <h2
          class="sectionTitle"
        >
          My Missions
        </h2>
        <p
          class="noItems"
        >
          You haven't yet become an
          <span
            class="badge misBadge misActive"
          >
            Active Member
          </span>
          of any Missions.
        </p>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
